src/annotation.js

```diff
@@ -45,11 +45,19 @@ var annotation = function (type, args) {
     return new annotation(type, args);
   }
 
-  annotationId += 1;
   var m_options = $.extend({}, {showLabel: true}, args || {}),
-      m_id = annotationId,
-      m_name = m_options.name || (
-        type.charAt(0).toUpperCase() + type.substr(1) + ' ' + annotationId),
+      m_id = m_options.annotationId;
+  delete m_options.annotationId;
+  if (m_id === undefined || (m_options.layer && m_options.layer.annotationById(m_id))) {
+    annotationId += 1;
```

Comment:
I wonder if we should emit a warning here if the "requested" id is ignored.  Even better, perhaps if we are wanting to preserve id's across contexts, maybe we should just make them uuid's.  I think the value is fairly opaque with the exception of the default title.

Also, the id attribute isn't documented.  Maybe it's better to just keep it as a private interface due to the id conflict issues.  Thoughts?

Reply:
The default title was my main motivation of using low integers.  We could use uuids for the ids, change the accessor function from `id` to `_id` to indicate that it is private, and use a different counter for generating mostly distinct names.

Reply:
I'm okay with leaving the switch to uuid until later.  I think it would be good to issue a warning here to avoid confusion given how Emory is using these id's.